### ISSUE TITLE
Java 1.8 - VM cluster build fails

### ISSUE THREAD

**Issue by bijugs** [bloomberg/chef-bach#586]
Tried to create a new VM cluster with ``default['java']['jdk_version']`` set to 8 and ``automated_install`` fails with the following error in worker node.

```
[2016-06-24T15:19:49-04:00] INFO: template[/etc/tez/conf/tez-site.xml] created file /etc/tez/conf/tez-site.xml
[2016-06-24T15:19:49-04:00] INFO: template[/etc/tez/conf/tez-site.xml] updated file contents /etc/tez/conf/tez-site.xml
[2016-06-24T15:19:49-04:00] INFO: template[/etc/tez/conf/tez-site.xml] mode changed to 655
[2016-06-24T15:19:49-04:00] INFO: Processing bash[make_apps_tez_dir] action run (bcpc-hadoop::tez line 32)

================================================================================
Error executing action `run` on resource 'bash[make_apps_tez_dir]'
================================================================================


Mixlib::ShellOut::ShellCommandFailed
------------------------------------
Expected process to exit with [0], but received '1'
---- Begin output of "bash"  "/tmp/chef-script20160624-1721-4wuqw8" ----
STDOUT:
STDERR: Exception in thread "main" java.lang.UnsupportedClassVersionError: org/apache/hadoop/fs/FsShell : Unsupported major.minor version 51.0
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:643)
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
        at java.net.URLClassLoader.defineClass(URLClassLoader.java:277)
        at java.net.URLClassLoader.access$000(URLClassLoader.java:73)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:212)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(URLClassLoader.java:205)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:323)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:296)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:268)
        at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:406)
```
```
[2016-06-24T15:19:59-04:00] INFO: [template[/etc/hadoop/conf/hadoop-env.sh]] sending restart action to service[hadoop-yarn-nodemanager] (delayed)
[2016-06-24T15:19:59-04:00] INFO: Processing service[hadoop-yarn-nodemanager] action restart (bcpc-hadoop::datanode line 329)

================================================================================
Error executing action `restart` on resource 'service[hadoop-yarn-nodemanager]'
================================================================================


Mixlib::ShellOut::ShellCommandFailed
------------------------------------
Expected process to exit with [0], but received '1'
---- Begin output of /etc/init.d/hadoop-yarn-nodemanager restart ----
STDOUT: * Stopping Hadoop nodemanager:
stopping nodemanager
 * Starting Hadoop nodemanager:
starting nodemanager, logging to /var/log/hadoop-yarn/yarn-yarn-nodemanager-bcpc-vm3.out
Exception in thread "main" java.lang.UnsupportedClassVersionError: org/apache/hadoop/yarn/server/nodemanager/NodeManager : Unsupported major.minor version 51.0
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:643)
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
        at java.net.URLClassLoader.defineClass(URLClassLoader.java:277)
        at java.net.URLClassLoader.access$000(URLClassLoader.java:73)
```

**Comment by amithkanand** [comment]
This is failing on 1.7 as well.

**Comment by http-418** [comment]
We set the bigtop JAVA_HOME to `/usr/lib/jvm/default-java` here:

https://github.com/bloomberg/chef-bach/blob/master/cookbooks/bcpc-hadoop/attributes/default.rb#L135-L136

On Ubuntu 12.04, this will always be a symlink to openjdk 6, which is not expected to work with hdp.  The `default-java` symlink is intended for use by the debian package scripts, not end-users, so it is not managed by the alternatives system.

We should update the attribute instead, and set it to `/usr/lbi/jvm/java-8-oracle-amd64`

**Comment by http-418** [comment]
Biju points out that the java cookbook will update this symlink to point to Oracle java if the node is re-cheffed.

Perhaps we should `include_recipe bcpc-hadoop::configs` in the top of the 'bcpc-hadoop::tez' recipe, in order to guarantee the symlink runs before we attempt to run any hdfs commands

**Comment by http-418** [comment]
I still don't understand why the java cookbook symlink update was happening AFTER the bcpc-hadoop::tez recipe.

In any case, updating the attribute solves the problem:

https://github.com/bloomberg/chef-bach/pull/616/commits/326e9381a5e095e2975964e12deb9958e62f54cb

**Comment by http-418** [comment]
This is fixed in PR #616
